lxl-web/src/lib/utils/holdings.ts

```diff
@@ -185,3 +203,163 @@ export function getMyLibsFromHoldings(
 	}
 	return Object.values(result);
 }
+
+export async function getFullHolderData(allHolders: DecoratedHolder[]): Promise<FullHolderBySigel> {
+	const holderBySigel: FullHolderBySigel = {};
+
+	for (const h of allHolders) {
+		const id = h.obj?.['@id'];
+		const libraryRes = await fetch(`${id}?framed=true`, {
+			headers: { Accept: 'application/ld+json' }
+		});
+		const resJson = await libraryRes.json();
+		const libraryMainEntity = resJson['mainEntity'] as FramedData;
+
+		if (libraryMainEntity) {
+			holderBySigel[h.sigel] = libraryMainEntity;
+		}
+	}
+	return holderBySigel;
+}
+
+export async function fetchHoldersIfAbsent(holdersByType: HoldersByType) {
+	const cachedHolders = holdersCache.holders;
+	const allHolders = Object.values(holdersByType).flat();
+	for (const h of allHolders) {
+		const id = h.obj?.['@id'];
+
+		if (h.sigel && cachedHolders && !cachedHolders[h.sigel]) {
+			const response = await fetch(`${id}?framed=true`, {
+				headers: { Accept: 'application/ld+json' }
+			});
+			if (response.ok) {
+				const resJson = await response.json();
+				const libraryMainEntity = resJson['mainEntity'] as FramedData;
+				if (libraryMainEntity) {
+					cachedHolders[h.sigel] = libraryMainEntity;
+				}
+			} else {
+				console.error(`Could not fetch holder data for ${id}`);
+			}
+		}
+	}
+}
```

Comment:
could be the same fetch?
error handling missing in the first version.

Reply:
The first function is unused and was probably used before adding the cache+error handling. Must have missed to delete it! Good catch